x-pack/plugins/ml/public/maps/util.ts

```diff
@@ -23,6 +23,33 @@ export const ML_ANOMALY_LAYERS = {
 } as const;
 
 export type MlAnomalyLayersType = typeof ML_ANOMALY_LAYERS[keyof typeof ML_ANOMALY_LAYERS];
+const INFLUENCER_LIMIT = 3;
+const INFLUENCER_MAX_VALUES = 3;
+
+function getInfluencersHtmlString(
+  influencers: Array<{ influencer_field_name: string; influencer_field_values: string[] }>,
+  splitFields: string[]
+) {
+  let htmlString = '<ul>';
+  for (let i = 0; i < influencers.length; i++) {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    const { influencer_field_name, influencer_field_values } = influencers[i];
+    // Skip if there are no values or it's a partition field
+    if (!influencer_field_values.length || splitFields.includes(influencer_field_name)) continue;
+
+    const fieldValuesString = influencer_field_values.slice(0, INFLUENCER_MAX_VALUES).join(', ');
+
+    htmlString += `<li>${influencer_field_name}: ${fieldValuesString}</li>`;
+
+    // Only show top two influencers
+    if (i === INFLUENCER_LIMIT - 1) {
+      htmlString += '</ul>';
+      break;
+    }
+  }
+
+  return htmlString;
```

Comment:
If influencers.length == 0, would this `htmlString` be `'<ul>'` which is invalid html 🤔 ?

Reply:
Good call - checked for influencer length before entering the function and also moved closing tag out of the loop in 1dd59765c032b18dd9de60c53f9a1c24e86cf5d4